packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,566 @@
+        nextCursor = response.next_cursor
+      } else {
+        hasNextPage = false
+      }
+    }
+    for (const txn of committedTransactions) {
+      try {
+        if (
+          txn.raw_hex !== undefined &&
+          txn.confirmed_at !== null &&
+          txn.intent_hash !== undefined &&
+          txn.confirmed_at !== undefined
+        ) {
+          const transaction: Tx = await this.convertTransactionFromHex(txn.raw_hex, txn.intent_hash, txn.confirmed_at)
+          txList.txs.push(transaction)
+        }
+      } catch (error) {}
+    }
+    return txList
+  }
+
+  /**
+   * Get the transaction details of a given transaction id.
+   * This method uses LTSRadixEngineToolkit.Transaction.summarizeTransaction
+   * to convert a transaction hex to a transaction summary. If the transaction was not built with
+   * the SimpleTransactionBuilder, the method will fail to get the transaction data
+   * @param {string} txId The transaction id.
+   * @returns {Tx} The transaction details of the given transaction id.
+   */
+  async getTransactionData(txId: string): Promise<Tx> {
+    try {
+      const transactionCommittedDetailsRequest: TransactionCommittedDetailsRequest = {
+        intent_hash: txId,
+        opt_ins: {
+          raw_hex: true,
+        },
+      }
+      const transactionCommittedDetailsResponse: TransactionCommittedDetailsResponse =
+        await this.gatewayApiClient.transaction.innerClient.transactionCommittedDetails({
+          transactionCommittedDetailsRequest: transactionCommittedDetailsRequest,
+        })
+      if (
+        transactionCommittedDetailsResponse.transaction.raw_hex !== undefined &&
+        transactionCommittedDetailsResponse.transaction.confirmed_at !== null &&
+        transactionCommittedDetailsResponse.transaction.confirmed_at !== undefined &&
+        transactionCommittedDetailsResponse.transaction.intent_hash !== undefined
+      ) {
+        const transaction: Tx = await this.convertTransactionFromHex(
+          transactionCommittedDetailsResponse.transaction.raw_hex,
+          transactionCommittedDetailsResponse.transaction.intent_hash,
+          transactionCommittedDetailsResponse.transaction.confirmed_at,
+        )
+        return transaction
+      } else {
+        throw new Error('Incomplete transaction data received')
+      }
+    } catch (error) {
+      throw new Error('Failed to fetch transaction data')
+    }
+  }
+
+  /**
+   * Helper function to convert a transaction in hex, returned by the gateway to a Tx type
+   * @param transaction_hex - The raw_hex returned by the gateway for a transaction id
+   * @param confirmed_at - The confirmed_at date for the transaction
+   * @param intent_hash - The transaction intent hash
+   * @returns a transaction in Tx type
+   */
+  async convertTransactionFromHex(transaction_hex: string, intent_hash: string, confirmed_at: Date): Promise<Tx> {
+    const binaryString = Buffer.from(transaction_hex, 'hex').toString('binary')
```

Comment:
You can just use `Convert.HexString.toUint8Array` here instead of this line and the one below.

Reply:
done